[kwangthiag] iP

kwangthiag opened the pull request:
# ~~Duke~~ Bot
>    “They don't know me son” – David Goggins

## Bot frees your mind of having to remember things you need to do. It's,

-     text-based
-     easy to learn
-     FAST SUPER FAST to use

## All you need to do is,

1.     download it from [here][(https://github.com/kwangthiag/ip/releases/tag/A-Jar)].
2.     double-click it.
3.     add your tasks.
4.     let it manage your tasks for you 😉

And it is **FREE!**

## Features:
- [x] Managing todo, event and deadline tasks
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

kwangthiag committed:
l0 done

kwangthiag committed:
l1 done

kwangthiag committed:
l1 done

kwangthiag committed:
l1 done

kwangthiag committed:
l2 done

kwangthiag committed:
l3 done

kwangthiag committed:
l3 done

kwangthiag committed:
l4 done

kwangthiag committed:
atextuitesting  done

kwangthiag committed:
l5 done

kwangthiag committed:
l6 done

kwangthiag committed:
Added Save function

kwangthiag committed:
Added Save function

kwangthiag committed:
Merge branch 'branch-level-7'

kwangthiag committed:
Fixed Error Handling and added Date and Time support

kwangthiag committed:
Merge branch-Level-8

kwangthiag committed:
Added OOP

kwangthiag committed:
Added OOP

kwangthiag committed:
Merge branch-AOOP

kwangthiag committed:
Added packages

kwangthiag committed:
Merge branch-Package

kwangthiag committed:
Merge remote-tracking branch 'origin/add-gradle-support'

kwangthiag committed:
JUnit Tests Added

kwangthiag committed:
Jar made

kwangthiag committed:
Fixed Coding standards

kwangthiag committed:
Added Find

kwangthiag committed:
Merge branch 'branch-Level-9'

kwangthiag committed:
Deconflicted the merges

kwangthiag committed:
Fixed find and some conflicts

ruishanteo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,140 @@
+    /**
+     * Represents the list of tasks.
+     */
+    private Tasklist todoList;
```
Would plural form on names representing a collection of objects be better?

longnguyentan replied:
I also agree with @ruishanteo with the plural form for `todoList`

ruishanteo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,140 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
```
I like that you have provided a lot of documentations for fields, methods and classes.

ruishanteo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,137 @@
+import duke.exceptions.DukeException;
+import duke.task.*;
+
+import java.io.*;
```
Should imported methods be explicitly stated instead of using wildcards?

ruishanteo commented on src/main/java/duke/exceptions/DukeDatabaseException.java:
```diff
@@ -0,0 +1,8 @@
+package duke.exceptions;
+
+public class DukeDatabaseException extends DukeException {
```
I think having header comments for the public class be useful.

ruishanteo reviewed (commented):
Your code is easy to read and well executed!

longnguyentan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,140 @@
+     */
+    private boolean respond(String s) {
+        StringBuilder str = new StringBuilder(s);
+        String check1 = "";
```
`check1` can have a better meaning

longnguyentan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,140 @@
+    private boolean respond(String s) {
+        StringBuilder str = new StringBuilder(s);
+        String check1 = "";
+        String check2 = "";
```
`check2` can have a better meaning

longnguyentan commented on src/main/java/duke/Storage.java:
Most variable and function names follow the guideline.

longnguyentan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,97 @@
+    /**
+     * Completion status of the task.
+     */
+    protected boolean done;
```
`done` should be `isDone`

longnguyentan reviewed (commented):
Overall is good 👍 Keep it up :)

kwangthiag committed:
Added GUI Support

kwangthiag committed:
Added GUI Support

kwangthiag committed:
Merge tag 'branch-Level-10'

kwangthiag committed:
Add assertions.

Add assert.isRunning to exit method in Duke.java.

kwangthiag committed:
Merge pull request #2 from kwangthiag/branch-A-Assertions

Add assertions.

kwangthiag committed:
Fix Code Quality

Abstract out addToList method in Task.java into addTodo, addEvent and
addDeadline methods. Change boolean name done to isDone

kwangthiag committed:
Fix Code Quality

Abstract out addToList method in Task.java into addTodo, addEvent and addDeadline methods. Change boolean name done to isDone

kwangthiag committed:
Merge pull request #3 from kwangthiag/branch-A-CodeQuality

Fix Code Quality

kwangthiag committed:
Add Detect Duplicate Support

kwangthiag committed:
Merge pull request #4 from kwangthiag/branch-C-DetectDuplicates

Add Detect Duplicate Support

kwangthiag committed:
Add User Guide and ui.png

kwangthiag committed:
Merge branch 'master' of https://github.com/kwangthiag/ip

kwangthiag committed:
Fix Ui.png
